modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc

```diff
@@ -122,17 +123,22 @@ common::Status PiecewiseJerkPathOptimizer::Process(
       }
     }
 
+    // updated cost function for path reference
+    std::vector<double> path_reference_l(path_boundary_size, 0.0);
+    bool is_valid_path_reference = path_data.is_valid_path_reference();
+    size_t path_reference_size = path_data.path_reference().size();
+
     if (path_boundary.label().find("regular") != std::string::npos &&
-        path_data.is_valid_path_reference()) {
+        is_valid_path_reference) {
       // when path reference is ready
-      // set end lateral to be at the path reference
-      common::SLPoint path_reference_end_sl;
-      reference_line.XYToSL(path_data.path_reference_end_pose(),
-                            &path_reference_end_sl);
-      end_state[0] = path_reference_end_sl.l();
```

Comment:
this line can't be deleted?
end_state[0] will be used by a check later on.

Reply:
Updated on line 142. Thank you.

Reply:
Added on line 142. Thank you.